federation/executor.go

```diff
@@ -0,0 +1,89 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/samsarahq/thunder/graphql"
+	"github.com/samsarahq/thunder/graphql/introspection"
+)
+
+type ExecutorClient interface {
+	Execute(ctx context.Context, req *graphql.Query) ([]byte, error)
+}
+
+// Executor has a map of all the executor clients such that it can execute a
+// subquery on any of the federated servers.
+// The planner allows it to coordinate the subqueries being sent to the federated servers
+type Executor struct {
+	Executors map[string]ExecutorClient
+	planner   *Planner
+}
+
+func fetchSchema(ctx context.Context, e ExecutorClient) ([]byte, error) {
+	query, err := graphql.Parse(introspection.IntrospectionQuery, map[string]interface{}{})
+	if err != nil {
+		return nil, err
+	}
+
+	return e.Execute(ctx, query)
```

Comment:
This is the default implementation that queries each of the executor clients to get their introspected schemas and create a merged schema. This would be used if you want to run the federated executor locally.

In production, however, we want to create a merged schema based on the different versions that are deployed and get the intersection. The logic for this is split up into a separate pr that will come shortly.